lldb/source/Plugins/ExpressionParser/Swift/SwiftExpressionParser.cpp

```diff
@@ -640,36 +648,42 @@ static void AddRequiredAliases(Block *block, lldb::StackFrameSP &stack_frame_sp,
     swift::BoundGenericEnumType *optional_type =
         referent_type->getAs<swift::BoundGenericEnumType>();
 
-    if (!optional_type)
-      return;
+    if (!optional_type || optional_type->getGenericArgs().empty())
+      return Status("Could not add required aliases: could not get optional "
+                    "type for self.");
 
     swift::Type first_arg_type = optional_type->getGenericArgs()[0];
 
-    swift::ClassType *self_class_type =
-        first_arg_type->getAs<swift::ClassType>();
-
-    if (!self_class_type)
-      return;
+    // Only class types can be weakly captured.
```

Comment:
```suggestion
    // The Swift expression parser can only weakly capture class types.
```

Reply:
In this case it's not the Swift expression parser, it's in the rules of the Swift language that only a class type can be weakly captured. Also, "strongly captured" and "weakly captured" are already part of the Swift lingo.

I'll change it to:

// In Swift only class types can be weakly captured.